pysb/simulator/cupsoda.py

```diff
@@ -250,58 +334,57 @@ def run(self, tspan=None, initials=None, param_values=None):
                                           param_values=param_values,
                                           _run_kwargs=[])
 
+        if isinstance(self.gpu, collections.Iterable):
```

Comment:
This kind of normalization probably belongs in `__init__` -- can you just apply this directly to `self.gpu` so it's always a list by the time the code gets to `run`?